PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:13 + add:0, change matches:
-    _inherit = 'mrp.bom'
-    _inherit = 'mrp.production'
-    _inherit = 'product.supplierinfo'
-    _inherit = 'product.product'
-    _inherit = 'res.company'
-    _inherit = 'res.partner'
-    _inherit = 'stock.location'
-    _inherit = 'stock.move'
-    _inherit = 'stock.move.line'
-    _inherit = 'stock.picking'
-    _inherit = 'stock.quant'
-    _inherit = "stock.rule"
-    _inherit = 'stock.warehouse'

Total Changes: 34

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/mrp_subcontracting/models/mrp_bom.py
+++ b/addons/mrp_subcontracting/models/mrp_bom.py
@@ -5,8 +5,9 @@ from odoo import api, fields, models, _
 from odoo.exceptions import ValidationError
 from odoo.osv.expression import AND
 
+
 class MrpBom(models.Model):
-    _inherit = 'mrp.bom'
+    _inherit = ['mrp.bom']
 
     type = fields.Selection(selection_add=[
         ('subcontract', 'Subcontracting')

--- a/addons/mrp_subcontracting/models/mrp_production.py
+++ b/addons/mrp_subcontracting/models/mrp_production.py
@@ -8,7 +8,7 @@ from odoo.tools.float_utils import float_compare, float_is_zero
 
 
 class MrpProduction(models.Model):
-    _inherit = 'mrp.production'
+    _inherit = ['mrp.production']
     _rec_names_search = ['name', 'incoming_picking.name']
 
     move_line_raw_ids = fields.One2many(

--- a/addons/mrp_subcontracting/models/product.py
+++ b/addons/mrp_subcontracting/models/product.py
@@ -4,8 +4,8 @@
 from odoo import api, fields, models
 
 
-class SupplierInfo(models.Model):
-    _inherit = 'product.supplierinfo'
+class ProductSupplierinfo(models.Model):
+    _inherit = ['product.supplierinfo']
 
     is_subcontractor = fields.Boolean('Subcontracted', compute='_compute_is_subcontractor', help="Choose a vendor of type subcontractor if you want to subcontract the product")
 
@@ -18,7 +18,7 @@ class SupplierInfo(models.Model):
 
 
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     def _prepare_sellers(self, params=False):
         if params and params.get('subcontractor_ids'):

--- a/addons/mrp_subcontracting/models/res_company.py
+++ b/addons/mrp_subcontracting/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
 
     subcontracting_location_id = fields.Many2one('stock.location')
 

--- a/addons/mrp_subcontracting/models/res_partner.py
+++ b/addons/mrp_subcontracting/models/res_partner.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     property_stock_subcontractor = fields.Many2one(
         'stock.location', string="Subcontractor Location", company_dependent=True,

--- a/addons/mrp_subcontracting/models/stock_location.py
+++ b/addons/mrp_subcontracting/models/stock_location.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class StockLocation(models.Model):
-    _inherit = 'stock.location'
+    _inherit = ['stock.location']
 
     is_subcontracting_location = fields.Boolean(
         "Is a Subcontracting Location?",

--- a/addons/mrp_subcontracting/models/stock_move.py
+++ b/addons/mrp_subcontracting/models/stock_move.py
@@ -10,7 +10,7 @@ from odoo.tools.misc import OrderedSet
 
 
 class StockMove(models.Model):
-    _inherit = 'stock.move'
+    _inherit = ['stock.move']
 
     is_subcontract = fields.Boolean('The move is a subcontract receipt')
     show_subcontracting_details_visible = fields.Boolean(

--- a/addons/mrp_subcontracting/models/stock_move_line.py
+++ b/addons/mrp_subcontracting/models/stock_move_line.py
@@ -6,7 +6,7 @@ from odoo import _, api, models
 
 
 class StockMoveLine(models.Model):
-    _inherit = 'stock.move.line'
+    _inherit = ['stock.move.line']
 
     @api.onchange('lot_name', 'lot_id')
     def _onchange_serial_number(self):

--- a/addons/mrp_subcontracting/models/stock_picking.py
+++ b/addons/mrp_subcontracting/models/stock_picking.py
@@ -10,8 +10,7 @@ from dateutil.relativedelta import relativedelta
 
 
 class StockPicking(models.Model):
-    _name = 'stock.picking'
-    _inherit = 'stock.picking'
+    _inherit = ['stock.picking']
 
     move_line_ids_without_package = fields.One2many(
         domain=['&', '|', ('location_dest_id.usage', '!=', 'production'), ('move_id.picking_code', '!=', 'outgoing'),

--- a/addons/mrp_subcontracting/models/stock_quant.py
+++ b/addons/mrp_subcontracting/models/stock_quant.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class StockQuant(models.Model):
-    _inherit = 'stock.quant'
+    _inherit = ['stock.quant']
 
     is_subcontract = fields.Boolean(store=False, search='_search_is_subcontract')
 

--- a/addons/mrp_subcontracting/models/stock_replenish_mixin.py
+++ b/addons/mrp_subcontracting/models/stock_replenish_mixin.py
@@ -4,8 +4,8 @@ from odoo import models, _
 from odoo.osv import expression
 
 
-class ProductReplenishMixin(models.AbstractModel):
-    _inherit = 'stock.replenish.mixin'
+class StockReplenishMixin(models.AbstractModel):
+    _inherit = ['stock.replenish.mixin']
 
     def _get_allowed_route_domain(self):
         domains = super()._get_allowed_route_domain()

--- a/addons/mrp_subcontracting/models/stock_rule.py
+++ b/addons/mrp_subcontracting/models/stock_rule.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockRule(models.Model):
-    _inherit = "stock.rule"
+    _inherit = ["stock.rule"]
 
     def _push_prepare_move_copy_values(self, move_to_copy, new_date):
         new_move_vals = super(StockRule, self)._push_prepare_move_copy_values(move_to_copy, new_date)

--- a/addons/mrp_subcontracting/models/stock_warehouse.py
+++ b/addons/mrp_subcontracting/models/stock_warehouse.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class StockWarehouse(models.Model):
-    _inherit = 'stock.warehouse'
+    _inherit = ['stock.warehouse']
 
     subcontracting_to_resupply = fields.Boolean(
         'Resupply Subcontractors', default=True)
